### PR TITLE
Partially revert Float16 availability changes

### PR DESCRIPTION
**Pull request by stephentyrone**
Specifically, when building for macOS/x86_64, use the old availability annotation instead of marking Float16 unconditionally unavailable to give downstream clients a window to adjust their own annotations. Update to https://github.com/apple/swift/pull/34821.

Addresses: <rdar://problem/71627096>

### CODE REVIEW

**Comment by stephentyrone** [comment]
@swift-ci please test source compatibility.

**Review by xedin — APPROVED** [review]

**Comment by xedin** [comment]
@swift-ci please test source compatibility release

**Comment by benlangmuir** [comment]
@swift-ci please test

File: stdlib/public/core/FloatingPointTypes.swift.gyb [hunk 1]
```diff
@@ -1352,7 +1352,13 @@ internal struct _${Self}AnyHashableBox: _AnyHashableBox {
 
 ${SelfDocComment}
 @frozen
+%  if bits == 16:
+@available(iOS 14, tvOS 14, watchOS 7, *)
+@available(OSX, unavailable)
```
Comment (xwu) [inline stdlib/public/core/FloatingPointTypes.swift.gyb:1357]: Nit:

```suggestion
@available(macOS, unavailable)
```

**Comment by stephentyrone** [comment]
@swift-ci please test

**Comment by shahmishal** [comment]
@swift-ci please test source compat

**Comment by stephentyrone** [comment]
@swift-ci please smoke test macos

**Comment by stephentyrone** [comment]
(Smoke testing macOS because we already had a clean test of this for macOS, and the failure we saw in the last run was an unrelated libc++ issue)

**Comment by shahmishal** [comment]
@swift-ci please smoke test macos

**Pull request merged**